recordm/customUI/dash/src/components/Calendar.vue

```diff
@@ -309,10 +375,99 @@
           this.calendarApi.changeView(newContent.activeView ? newContent.activeView : this.calendarOptions.initialView )
         }
       },
+      getCustomButtomObject(viewThisObject){
+        if (viewThisObject.calendarOptions["customButtons"]){
+          return viewThisObject.calendarOptions["customButtons"]
+        }
+        function dateChanges(prev,isToday) {
+          const previousDateElement = viewThisObject.datePickerElement.children[0]
+          const nextDateElement = viewThisObject.datePickerElement.children[2]
+
+          let prevDate, nextDate
+
+          prevDate = new Date(previousDateElement.value)
+          nextDate = new Date(nextDateElement.value)
+          
+          let diff = nextDate.getTime() - prevDate.getTime()
 
+          if(diff == 0){
+            diff = nextDate.getTime() - (new Date(nextDate.getFullYear(),nextDate.getMonth(),nextDate.getDate()-1))
+          }
+
+          diff = prev ? diff*(-1):diff
+
+          prevDate = new Date(prevDate.getTime()+diff)
+          nextDate = new Date(nextDate.getTime()+diff)
+
+          if(isToday){
+            prevDate = new Date()
+            nextDate = new Date(prevDate.getFullYear()+1,nextDate.getMonth(),nextDate.getDate())
+          } 
+          previousDateElement.value = getInputTypeOfDate(prevDate)
+          nextDateElement.value = getInputTypeOfDate(nextDate)
+
+          viewThisObject.dateRange = [prevDate, nextDate]
+        }
+        
+        return {
+          prev: {
+            text: 'prev',
+            click: function() {
+              dateChanges(true,false)
+            }
+          },
+          next: {
+            text: 'next',
+            click: function() {
+              dateChanges(false,false)
+            }
+          },
+          today: {
+            text: "today",
+            click: function () {
+              dateChanges(false,true)
+            }
+          }
+        }
+      },
+      handleListYear(){
+        if(this.calendarApi && this.calendarApi.view.type == "listYear"){
+          this.listYearSelected = true
+          this.calendarOptions["customButtons"] = this.getCustomButtomObject(this)
```

Comment:
Acho que a função devia-se chamar `this.getListYearButtons(this)`

Reply:
Porque ? A função tem acesso ao this

Reply:
Acho que o comentário não foi claro ou então não entendi a tua pergunta. O que eu estou a dizer é que a função `getCustomButtomObject` devia-se chamar `getListYearButtons`.

Reply:
Okk, percebi